components/uniset/src/enum_props.rs

```diff
@@ -4,672 +4,260 @@
-    DecompositionType,
-    EastAsianWidth,
-    GeneralCategory,
-    GraphemeClusterBreak,
-    HangulSyllableType,
-    IndicPositionalCategory,
-    IndicSyllabicCategory,
-    JoiningGroup,
-    JoiningType,
-    LineBreak,
-    LeadCanonicalCombiningClass,
-    NFCQuickCheck,
-    NFDQuickCheck,
-    NFKCQuickCheck,
-    NFKDQuickCheck,
-    NumericType,
-    SentenceBreak,
-    TrailCanonicalCombiningClass,
-    VerticalOrientation,
-    WordBreak,
-}
-
-#[derive(Clone, PartialEq, Debug)]
-pub enum BidiClass {
-    ArabicLetter,
-    ArabicNumber,
-    ParagraphSeparator,
-    BoundaryNeutral,
-    CommonSeparator,
-    EuropeanNumber,
-    EuropeanSeparator,
-    EuropeanTerminator,
-    FirstStrongIsolate,
-    LeftToRight,
-    LeftToRightEmbedding,
-    LeftToRightIsolate,
-    LeftToRightOverride,
-    NonspacingMark,
-    OtherNeutral,
-    PopDirectionalFormat,
-    PopDirectionalIsolate,
-    RightToLeft,
-    RightToLeftEmbedding,
-    RightToLeftIsolate,
-    RightToLeftOverride,
-    SegmentSeparator,
-    WhiteSpace,
-}
-
-#[derive(Clone, PartialEq, Debug)]
-pub enum BidiPairedBracketType {
-    Close,
-    None,
-    Open,
-}
-
-// The Rust enum determinant values are meaningful in ICU4C for collation
-// implementation purposes, so those numerical values are preserved here, too.
-#[derive(Clone, PartialEq, Debug)]
-pub enum CanonicalCombiningClass {
-    NotReordered = 0,
-    Overlay = 1,
-    CCC10 = 10,
-    CCC103 = 103,
-    CCC107 = 107,
-    CCC11 = 11,
-    CCC118 = 118,
-    CCC12 = 12,
-    CCC122 = 122,
-    CCC129 = 129,
-    CCC13 = 13,
-    CCC130 = 130,
-    CCC132 = 132,
-    CCC133 = 133,
-    CCC14 = 14,
-    CCC15 = 15,
-    CCC16 = 16,
-    CCC17 = 17,
-    CCC18 = 18,
-    CCC19 = 19,
-    CCC20 = 20,
-    AttachedBelowLeft = 200,
-    AttachedBelow = 202,
-    CCC21 = 21,
-    AttachedAbove = 214,
-    AttachedAboveRight = 216,
-    BelowLeft = 218,
-    CCC22 = 22,
-    Below = 220,
-    BelowRight = 222,
-    Left = 224,
-    Right = 226,
-    AboveLeft = 228,
-    CCC23 = 23,
-    Above = 230,
-    AboveRight = 232,
-    DoubleBelow = 233,
-    DoubleAbove = 234,
-    CCC24 = 24,
-    IotaSubscript = 240,
-    CCC25 = 25,
-    CCC26 = 26,
-    CCC27 = 27,
-    CCC28 = 28,
-    CCC29 = 29,
-    CCC30 = 30,
-    CCC31 = 31,
-    CCC32 = 32,
-    CCC33 = 33,
-    CCC34 = 34,
-    CCC35 = 35,
-    CCC36 = 36,
-    HanReading = 6,
-    Nukta = 7,
-    KanaVoicing = 8,
-    CCC84 = 84,
-    Virama = 9,
-    CCC91 = 91,
-}
-
-#[derive(Clone, PartialEq, Debug)]
-pub enum DecompositionType {
-    Can,
-    Com,
-    Enc,
-    Fin,
-    Font,
-    Fra,
-    Init,
-    Iso,
-    Med,
-    Nar,
-    Nb,
-    None,
-    Sml,
-    Sqr,
-    Sub,
-    Sup,
-    Vert,
-    Wide,
-}
-
-#[derive(Clone, PartialEq, Debug)]
-pub enum EastAsianWidth {
-    Ambiguous,
-    Fullwidth,
-    Halfwidth,
-    Neutral,
-    Narrow,
-    Wide,
+    GeneralCategory = 0x1005,
+    Script = 0x100A,
+    ScriptExtensions = 0x7000,
```

Comment:
Is it on purpose that this PR reduces the set of enumerated properties from a couple dozen to three?

Reply:
Yes. There are 3 different related issues that were easier to handle with 1 PR rather than artificially split them up and create rebase churn, so this PR became it. This PR is titled based on its original issue #537, #536 is related, and #588 is the issue that reduces the number of enumerated/catalog properties to match those needed for ECMA-262 regexp for the time being.